grails-app/services/au/org/ala/volunteer/TaskService.groovy

```diff
@@ -1557,6 +1557,25 @@ ORDER BY record_idx, name;
         lastActivities
     }
 
+    /**
+     * Returns a map of user activity between two given dates.
+     * @param startDate the start date
+     * @param endDate the end date
+     * @return a Map of user activity for a given date range.
+     */
+    def getUserActivityBetweenDates(Date startDate, Date endDate) {
+        String query = """\
```

Comment:
Seems like you don't need to select `to_timestamp(max(vt.lastView)/1000)` as it's not used?

Also this could be done with jooq for typesafe queries, etc?

Reply:
This was in case Alex wanted date parameters on the existing User summary report, in the end he decided against. I'm tempted to leave it in if that requirement changes, if you're happy with it.